### PR TITLE
feat(conditions): add schedule condition

### PR DESCRIPTION
**Pull request by sileht**
Change-Id: Ia58ae9db00a50997954346435314b7231363f158

### CODE REVIEW

**Comment by sileht** [comment]
This pull request is part of a stack:
1. feat(rules): allow and/or in conditions ([#2691](https://github.com/Mergifyio/mergify-engine/pull/2691))
1. feat: allow merged/closed/author/base/head in and/or conditions ([#2692](https://github.com/Mergifyio/mergify-engine/pull/2692))
1. feat(conditions): allow checks condition in and/or ([#2693](https://github.com/Mergifyio/mergify-engine/pull/2693))
1. refactor(filter): make it generic ([#2694](https://github.com/Mergifyio/mergify-engine/pull/2694))
1. feat(time): add time conditions ([#2695](https://github.com/Mergifyio/mergify-engine/pull/2695))
1. feat(conditions): add day/month/year/day-of-week conditions ([#2705](https://github.com/Mergifyio/mergify-engine/pull/2705))
1. feat(conditions): add support for named day-of-week ([#2708](https://github.com/Mergifyio/mergify-engine/pull/2708))
1. feat(conditions): add calendar condition ([#2709](https://github.com/Mergifyio/mergify-engine/pull/2709))

File: docs/source/conditions.rst [hunk 1]
```diff
@@ -279,6 +282,7 @@ Here's the list of pull request attribute that can be used in conditions:
    never be true.
 
    ``current-time`` does not support ``~=``, ``=`` and ``!=`` operators.
+   ``calendar`` only support ``=`` and ``!=``
```
Comment (jd) [inline docs/source/conditions.rst:285]:
```suggestion
   ``current-time`` does not support ``~=``, ``=`` and ``!=`` operators.
   ``schedule`` only supports ``=`` and ``!=``.
```

File: docs/source/conditions.rst [hunk 2]
```diff
@@ -270,6 +270,9 @@ Here's the list of pull request attribute that can be used in conditions:
        ``Mon``, ``Tue``, ``Wed``, ``Thu``, ``Fri``, ``Sat``, ``Sun``
        ``Monday``, ``Tuesday``, ``Wednesday``, ``Thursday``, ``Friday``,
        ``Saturday``, ``Sunday``. Weeks start on Monday.
+   * - ``calendar``
+     - string
+     - A calendar, e.g.: ``Mon-Fri``, ``12:00-18:00`` or ``Mon-Fri 12:00-18:00``
```
Comment (jd) [inline docs/source/conditions.rst:275]:
```suggestion
   * - ``schedule``
     - string
     - A calendar schedule, e.g., ``Mon-Fri``, ``12:00-18:00`` or ``Mon-Fri 12:00-18:00``.
```

File: mergify_engine/rules/filter.py [hunk 3]
```diff
@@ -80,6 +80,7 @@ def _identity(value: _T) -> _T:
         ">=": TreeBinaryLeafT,
         "!=": TreeBinaryLeafT,
         "~=": TreeBinaryLeafT,
+        "@": typing.Any,
```
Comment (jd) [inline mergify_engine/rules/filter.py:83]: - typing?
- add a comment about what this operator does

**Review by jd — CHANGES_REQUESTED** [review]

File: docs/source/conditions.rst [hunk 4]
```diff
@@ -270,6 +270,9 @@ Here's the list of pull request attribute that can be used in conditions:
        ``Mon``, ``Tue``, ``Wed``, ``Thu``, ``Fri``, ``Sat``, ``Sun``
        ``Monday``, ``Tuesday``, ``Wednesday``, ``Thursday``, ``Friday``,
        ``Saturday``, ``Sunday``. Weeks start on Monday.
+   * - ``schedule``
+     - string
+     - A schedule, e.g., ``Mon-Fri``, ``12:00-18:00`` or ``Mon-Fri 12:00-18:00``
```
Comment (jd) [inline docs/source/conditions.rst:275]:
```suggestion
     - A schedule, e.g., ``Mon-Fri``, ``12:00-18:00`` or ``Mon-Fri 12:00-18:00``.
```

**Review by jd — APPROVED** [review]

**Pull request merged**